src/main/java/seedu/address/model/pricerange/PriceRange.java

```diff
@@ -45,6 +42,11 @@ public static boolean isValidPriceRange(String test) {
 
         boolean isValid = test.matches(VALIDATION_REGEX);
 
+        // to prevent out of bounds error below
+        if (!isValid) {
+            return false;
+        }
+
         String[] rangeValues = test.split("-");
         double leftValue = Float.parseFloat(rangeValues[0].trim());
```

Comment:
It doesn't cause it to crash or anything because I think parseFloat just parses it into an Infinity number if it's beyond the range, but I realised that it actually affects the accuracy a bit? Like when I tested 20000000000000 as the upper boundary, it displayed as 199999999981808 (something like that). So I changed it to parseDouble and it seems to work now!